### PR TITLE
chore(data-warehouse): delete data from previous job using old paths

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- old data is hanging

## Changes

- delete it

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by EDsCODE** [comment]
@Gilbert09 Here's a possible way to do delete the data using the old folder paths

File: posthog/warehouse/models/external_data_job.py [hunk 1]
```diff
@@ -56,6 +63,19 @@ def get_external_data_job(job_id: UUID) -> ExternalDataJob:
...
+def aget_external_data_jobs_by_schema_id(schema_id: UUID) -> list[ExternalDataJob]:
+    from posthog.warehouse.models import ExternalDataSchema
+
+    return list(
+        ExternalDataJob.objects.prefetch_related(
+            "pipeline", Prefetch("schema", queryset=ExternalDataSchema.objects.prefetch_related("source"))
+        )
+        .filter(schema_id=schema_id)
+        .all()
+    )
```
Comment (Gilbert09) [inline posthog/warehouse/models/external_data_job.py:76]: We wanna delete data from all jobs except for the latest - someone may have synced their data and then turned off the jobs, and so we don't wanna delete that folder (even though it's using the old folder path)

Comment (EDsCODE) [reply posthog/warehouse/models/external_data_job.py:76]: ah yep, good catch

**Review by Gilbert09 — APPROVED** [review]
Great usage of the end to end tests!

**Pull request merged**